osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/exceptionhandling/FunctionalExceptionType.java

```diff
@@ -35,6 +35,19 @@ public enum FunctionalExceptionType {
   GATEWAY_DEVICE_NOT_SET_FOR_MBUS_DEVICE(220, "GATEWAY_DEVICE_NOT_SET_FOR_MBUS_DEVICE"),
   GATEWAY_DEVICE_INVALID_FOR_MBUS_DEVICE(221, "GATEWAY_DEVICE_INVALID_FOR_MBUS_DEVICE"),
   MBUS_DEVICE_NOT_MOVED_TO_ANOTHER_EMETER(222, "MBUS_DEVICE_NOT_MOVED_TO_ANOTHER_EMETER"),
+  AUTHENTICATION_ERROR(223, "AUTHENTICATION_ERROR"),
+  AUTHENTICATION_REQUIRED(224, "AUTHENTICATION_REQUIRED"),
+  CONNECTION_REFUSED(225, "CONNECTION_REFUSED"),
+  CONNECTION_RESET(226, "CONNECTION_RESET"),
+  CONNECTION_TIMED_OUT(227, "CONNECTION_TIMED_OUT"),
+  CONNECTION_ESTABLISH_ERROR(228, "CONNECTION_ESTABLISH_ERROR"),
+  SOCKET_CLOSED_BY_REMOTE(229, "SOCKET_CLOSED_BY_REMOTE"),
+  UNABLE_TO_DECYPHER(230, "UNABLE_TO_DECYPHER"),
+  WRAPPER_HEADER_ERROR(231, "WRAPPER_HEADER_ERROR"),
+  ILLEGAL_RESPONSE(232, "ILLEGAL_RESPONSE"),
+  RESPONSE_TIMEOUT(233, "RESPONSE_TIMEOUT"),
+  UNKNOWN_ASSOCIATION_RESULT(234, "UNKNOWN_ASSOCIATION_RESULT"),
+  UNKNOWN(235, "UNKNOWN"),
```

Comment:
Why do we define an Unknown connection error instead of using the general connection_error?

Reply:
Indeed, UNKNOWN is redundant. Removed.